src/coreclr/vm/ilstubcache.cpp

```diff
@@ -501,42 +484,28 @@ MethodDesc* ILStubCache::GetStubMethodDesc(
 
         if (SF_IsSharedStub(dwStubFlags))
         {
-            size_t cbSizeOfBlob = pParams->m_cbSizeOfBlob;
+            size_t cbSizeOfBlob = pHashBlob->m_cbSizeOfBlob;
             AllocMemHolder<ILStubHashBlob> pBlobHolder( m_heap->AllocMem(S_SIZE_T(cbSizeOfBlob)) );
```

Comment:
Let's move this into the `if (phe == NULL)` branch. I don't see any reason to allocate this prior to looking up what might already exist.